tests/std/tests/P1223R5_ranges_alg_find_last/test.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+using P = pair<int, int>;
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::find_last(borrowed<false>{}, 42)), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::find_last(borrowed<true>{}, 42)), ranges::subrange<int*>>);
+
+template <class T, class U>
+constexpr void check_value(const T& found, const U& value) {
+    if constexpr (same_as<T, P>) {
+        assert(found.first == value);
+        assert(found.second == 1729);
+    } else {
+        assert(found.peek().first == value);
+        assert(found.peek().second == 1729);
+    }
+}
+
+struct instantiator {
+    static constexpr P haystack[6] = {{0, 42}, {2, 42}, {4, 42}, {0, 1729}, {2, 1729}, {4, 1729}};
+
+    template <ranges::forward_range Read>
+    static constexpr void call() {
+        using ranges::find_last, ranges::common_range, ranges::iterator_t;
+
+        for (const auto& [value, _] : haystack) {
+            { // Validate range overload [found case]
+                Read wrapped_input{haystack};
+                const auto result = find_last(wrapped_input, value, get_first);
+                STATIC_ASSERT(same_as<iterator_t<decltype(result)>, iterator_t<Read>>);
+                STATIC_ASSERT(common_range<decltype(result)>);

```

Comment:
This could more concisely and precisely be:
```suggestion
                STATIC_ASSERT(same_as<decltype(result), const ranges::subrange<iterator_t<Read>>>);
```
(Applies several places in all three tests.)